### PR TITLE
feat(ui): Add sorting and pagination to performance metrics table [INGEST-544]

### PR DESCRIPTION
**Pull request by matejminar**
https://user-images.githubusercontent.com/9060071/150529643-10ac31dc-5a30-4b0a-9d72-33a1f9bf7df8.mp4

Backend PR: https://github.com/getsentry/sentry/pull/31181

### CODE REVIEW

File: static/app/utils/metrics/metricsRequest.tsx [hunk 1]
```diff
@@ -184,31 +206,34 @@ class MetricsRequest extends React.Component<Props, State> {
...
     }
   };
 
   render() {
-    const {reloading, errored, error, response, responsePrevious} = this.state;
+    const {reloading, errored, error, response, responsePrevious, pageLinks} = this.state;
     const {children, isDisabled, includeTabularData} = this.props;
 
-    const loading = response === null && !isDisabled;
+    const loading = response === null && !isDisabled && !error;
```
[the remark this answers is not in the document]
Comment (matejminar) [reply static/app/utils/metrics/metricsRequest.tsx:226]: Yes, if the first request fails, the loader would not stop spinning.

**Comment by jjbayer** [comment]
I noticed two things while testing (see screenshots):
1. p50 shows `n/a` even though the API seems to return a value
2. tpm shows `5.0 ms`, that's the wrong unit right? Also, we should divide by the number of minutes (I know we will move this arithmetic to the backend at some point).
Screen Shot 2022-01-24 at 17 10 50
Screen Shot 2022-01-24 at 17 12 48

File: static/app/utils/metrics/metricsRequest.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply static/app/utils/metrics/metricsRequest.tsx:226]: hm I would expect some error message

**Comment by matejminar** [comment]
@jjbayer both of these are caused by the overly simplistic map we do between metrics and discover fields: https://github.com/getsentry/sentry/blob/0fab3b22f359bb2c7f4854109dbacb459d37e602/static/app/utils/metrics/fields.tsx#L71
We could fix that but we would need to introduce additional frontend complexity that would be soon scraped and would not necessarily help us with testing the tabular API endpoint with pagination and sorting.

**Review by jjbayer — APPROVED** [review]
> We could fix that but we would need to introduce additional frontend complexity that would be soon scraped and would not necessarily help us with testing the tabular API endpoint with pagination and sorting.

@matejminar Alright, let's leave that as it is then. I tested with the table on the "Frontend" tab and everything works as expected!

**Comment by Metan6666** [comment]
Add to mrahmatipuia@gmail.com

**Pull request merged**